### ISSUE TITLE
A sniff to detect not efficient @phpcsSuppress annotations

### ISSUE THREAD

**Issue by soullivaneuh** [slevomat/coding-standard#1167]
# Summary

On the following example:

```php
/**
 * @phpcsSuppress SlevomatCodingStandard.Functions.UnusedParameter
 */
public function supports(Request $request)
{
    return $request->attributes->get('_route') === 'connect_check';
}
```

You can see the `@phpcsSuppress` is useless here because the `$request` variable is being used.

I use this annotation generally for vendor class extension because I can not remove them.

However, I might update my method override and then use the given parameter.

It would be a nice addition if a sniff can detect that kind of behavior and consider this as an  error.
We also may imagine the sniff being able to automatically remove the useless annotations.

I did some search on the issue tracker, but I didn't find any similar request.

What do you think?

**Comment by lkraav** [comment]
Yep, staleness detection sure would be nice.

**Comment by kukulich** [comment]
I would be difficult to create one sniff. It should be possible to add new error to each sniff...

**Comment by kukulich** [comment]
New report implemented for `UnusedParameterSniff`: https://github.com/slevomat/coding-standard/commit/060e1790ba3d42d78ea02d6a596d64c1e6265a73
